src/levanter/models/olmo.py

```diff
@@ -331,8 +333,10 @@ def __call__(self, x: NamedArray, mask: Optional[NamedArray | AttentionMask], *,
         v = v.rearrange((..., "kv_heads", "position", "head_size"))
 
         # Apply rotary position embeddings
-        rot_embs = self.config.rope.build(self.config.HeadSize, q.resolve_axis("position"))
-        q, k = rot_embs(self.config.HeadSize, q, k)
+        pos_ids = pos_ids or hax.arange(x.resolve_axis("position"))
+        rot_embs = self.config.rope.build(self.config.HeadSize)
```

Comment:
Same comment on moving this build function to init since it's static at inference time.

Reply:
(Side note: should we have a TODO to make OlmoAttention inherit from attention config?)

Reply:
honestly i'd rather not to keep the generic Attention from getting too unwieldy. better to do some copy/paste